src/platform/Linux/ConnectivityManagerImpl.cpp

```diff
@@ -66,5 +83,196 @@ void ConnectivityManagerImpl::_OnPlatformEvent(const ChipDeviceEvent * event)
+    {
+        ChipLogProgress(DeviceLayer, "wpa_supplicant: _IsWiFiStationConnected: interface not connected");
+        return false;
+    }
+
+    state = wpa_fi_w1_wpa_supplicant1_interface_get_state(mWpaSupplicant.iface);
+    ChipLogProgress(DeviceLayer, "wpa_supplicant: wpa_fi_w1_wpa_supplicant1_interface_get_state: %s", state);
+
+    if (g_strcmp0(state, "completed") == 0)
+    {
+        ChipLogProgress(DeviceLayer, "wpa_supplicant: WiFi already connected");
+        SetFlag(mConnectivityFlag, kFlag_HaveIPv4InternetConnectivity, true);
+        SetFlag(mConnectivityFlag, kFlag_HaveIPv6InternetConnectivity, true);
+        ret = true;
+    }
+
+    return ret;
+}
+
+bool ConnectivityManagerImpl::_CanStartWiFiScan()
+{
+    return mWpaSupplicant.state == GDBusWpaSupplicant::WPA_INTERFACE_CONNECTED &&
+        mWpaSupplicant.scanState == GDBusWpaSupplicant::WIFI_SCANNING_IDLE;
+}
+
+void ConnectivityManagerImpl::_OnWpaInterfaceProxyReady(GObject * source_object, GAsyncResult * res, gpointer user_data)
+{
+    GError * err                            = NULL;
+    WpaFiW1Wpa_supplicant1Interface * iface = wpa_fi_w1_wpa_supplicant1_interface_proxy_new_for_bus_finish(res, &err);
+
+    if (mWpaSupplicant.iface)
+    {
+        g_object_unref(mWpaSupplicant.iface);
+        mWpaSupplicant.iface = NULL;
+    }
+
+    if (iface != NULL && err == NULL)
+    {
+        mWpaSupplicant.iface = iface;
+        mWpaSupplicant.state = GDBusWpaSupplicant::WPA_INTERFACE_CONNECTED;
+        ChipLogProgress(DeviceLayer, "wpa_supplicant: connected to wpa_supplicant interface proxy: %p", iface);
+    }
+    else
+    {
+        if (err != NULL)
+            ChipLogProgress(DeviceLayer, "wpa_supplicant: failed to create wpa_supplicant1 interface proxy %s: %s",
+                            mWpaSupplicant.interfacePath, err->message);
+        else
+            ChipLogProgress(DeviceLayer, "wpa_supplicant: failed to create wpa_supplicant1 interface proxy %s",
+                            mWpaSupplicant.interfacePath);
+
+        mWpaSupplicant.state = GDBusWpaSupplicant::WPA_NOT_CONNECTED;
+    }
+
+    if (err != NULL)
+        g_error_free(err);
+}
+
+void ConnectivityManagerImpl::_OnWpaInterfaceReady(GObject * source_object, GAsyncResult * res, gpointer user_data)
+{
+    GError * err = NULL;
+
+    gboolean result =
+        wpa_fi_w1_wpa_supplicant1_call_get_interface_finish(mWpaSupplicant.proxy, &mWpaSupplicant.interfacePath, res, &err);
+    if (result)
+    {
+        mWpaSupplicant.state = GDBusWpaSupplicant::WPA_GOT_INTERFACE_PATH;
+        ChipLogProgress(DeviceLayer, "wpa_supplicant: WiFi interface: %s", mWpaSupplicant.interfacePath);
+
+        wpa_fi_w1_wpa_supplicant1_interface_proxy_new_for_bus(G_BUS_TYPE_SYSTEM, G_DBUS_PROXY_FLAGS_NONE, "fi.w1.wpa_supplicant1",
+                                                              mWpaSupplicant.interfacePath, NULL, _OnWpaInterfaceProxyReady, NULL);
+    }
+    else
+    {
+        if (err != NULL)
```

Comment:
Suggest removing this branch (see above)

Reply:
Ack